### PR TITLE
Monte Carlo Functionality

### PR DESCRIPTION
**Pull request by sgill2**
This PR implements the ability to perform Monte Carlo (MC) simulations (as opposed to NCMC), with tests.
In addition to MC simulations this PR:
1. Disables the alchemical dispersion correction for alchemical simulations when setting up the system with `openmmtools.alchemy`, greatly speeding up the NCMC simulation portion.
2. Adds a sanity check to the nstepsNC, which sets the number of NC steps to use in a NCMC simulation. Specifically blues.simulation.runNCMC() will throw a ValueError if nstepsNC is not even or nstepsNC < 1

### CODE REVIEW

**Comment by davidlmobley** [comment]
So, this would fix #82 ? 

What's the mechanism/framework for doing MC simulations? e.g., what would the user do differently?

**Comment by sgill2** [comment]
>So, this would fix #82 ?

Yes it would.

>What's the mechanism/framework for doing MC simulations? e.g., what would the user do differently?

From a user standpoint they would simply need to call Simulation.runMC() instead of Simulation.runNCMC() to run a MC+MD simulation. 
From a code perspective, runMC() is structured very similarly to runNCMC(), except that it uses a different acceptance criteria, and instantaneous move proposal instead of halfway through the NCMC protocol.

File: blues/tests/test_mc.py [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+            else:
+                self.reject += 1
+                print('NCMC MOVE REJECTED: log_ncmc {} < {}'.format(log_ncmc, randnum) )
+                self.md_sim.context.setPositions(md_state0['positions'])
+            self.log_ncmc = log_ncmc
+            self.md_sim.context.setVelocitiesToTemperature(self.temperature)
+        asim.acceptRejectMC = nacceptRejectMC
+        nacceptRejectMC.__get__(asim)
+        asim.acceptRejectMC = types.MethodType(nacceptRejectMC, asim)
+        asim.runMC()
```
Comment (davidlmobley) [inline blues/tests/test_mc.py]: So I'm confused. Why do we have `runMC` in this test, but much of the code just prior to this line talks about NCMC? Did you just forget to update the code when copying it from somewhere else? e.g., there is a print statement that says that the NCMC move was accepted, etc.

**Comment by davidlmobley** [comment]
Mostly, this looks good to me, @sgill2 , though you have a conflict which needs resolving, and you still have [WIP] in the title so I'm unclear if you're done with it. LMK if you're done and I can finish reviewing.

File: blues/tests/test_mc.py (hunk 1 above)
Comment (sgill2) [inline blues/tests/test_mc.py]: Opps, yeah. I copied the basic structure from another parts of the code, which came with NCMC references.

Comment (davidlmobley) [inline blues/tests/test_mc.py]: Great. I'm gonna have to start teasing you about this one, as it seems like this mistake gets you every time.

**Comment by sgill2** [comment]
@davidlmobley This should be about ready for review too.

**Comment by sgill2** [comment]
I'm not the best about remembering to remove the [WIP] tag when things are about done; I'll have to work on that.

File: blues/simulation.py [hunk 2]
```diff
@@ -433,21 +445,75 @@ def simulateMD(self):
...
+        log_ncmc = (md_state1['potential_energy'] - md_state0['potential_energy']) * (-1.0/self.nc_integrator.kT)
+        randnum =  math.log(np.random.random())
+
+        if log_ncmc > randnum:
+            self.accept += 1
+            print('NCMC MOVE ACCEPTED: log_ncmc {} > randnum {}'.format(log_ncmc, randnum) )
+            self.md_sim.context.setPositions(md_state1['positions'])
+        else:
+            self.reject += 1
+            print('NCMC MOVE REJECTED: log_ncmc {} < {}'.format(log_ncmc, randnum) )
```
Comment (davidlmobley) [inline blues/simulation.py]: You've still got references to NCMC here in your MC stuff -- "NCMC move rejected", "log_ncmc", etc. Have some pride in your work -- make sure the MC part only uses "MC" not "NCMC". Make it look nice. :man_in_tuxedo:

File: blues/tests/test_mc.py [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+        self.model.calculateProperties()
+        self.initial_positions = self.nc_sim.context.getState(getPositions=True).getPositions(asNumpy=True)
+        asim = Simulation(sims, self.mover, **self.opt)
+        #monkeypatch to access log_ncmc acceptance value
+        def nacceptRejectMC(self):
+            """Function that chooses to accept or reject the proposed move.
+            """
+            md_state0 = self.current_state['md']['state0']
+            md_state1 = self.current_state['md']['state1']
+            log_ncmc = (md_state1['potential_energy'] - md_state0['potential_energy']) * (-1.0/self.nc_integrator.kT)
```
Comment (davidlmobley) [inline blues/tests/test_mc.py]: :man_in_tuxedo: "log_ncmc"

File: blues/tests/test_mc.py [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+
+            if log_ncmc > randnum:
+                self.accept += 1
+                print('MC MOVE ACCEPTED: log_ncmc {} > randnum {}'.format(log_ncmc, randnum) )
+                self.md_sim.context.setPositions(md_state1['positions'])
+            else:
+                self.reject += 1
+                print('MC MOVE REJECTED: log_ncmc {} < {}'.format(log_ncmc, randnum) )
+                self.md_sim.context.setPositions(md_state0['positions'])
+            self.log_ncmc = log_ncmc
```
Comment (davidlmobley) [inline blues/tests/test_mc.py]: :man_in_tuxedo:

File: blues/tests/test_mc.py [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+from simtk import openmm
+from openmmtools import testsystems
+import types
+import math
+
+class BLUESTester(unittest.TestCase):
+    """
+    Test the Simulation class.
+    """
+    def setUp(self):
```
Comment (davidlmobley) [inline blues/tests/test_mc.py]: Do you want to reduce the redundancy between this and the `test_blues` module by, maybe, putting them into the same test script and then making a function both can call that does some of the setup common to both? I'm just wondering if there's an easy way to reduce redundancy so there is not as much to update when we want to change something in these tests. Your call on this one.

Comment (sgill2) [inline blues/tests/test_mc.py]: Combining the tests to use the same setup would be helpful. But in this case I'm using a different system in the tests (alanine in vacuum vs explicit solvent), and actually running some simulation steps for the vacuum system, so I'm leaning towards keeping them separate.

**Pull request merged**